packages/teleport/src/Discover/LoginTrait/useLoginTrait.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect } from 'react';
+import useAttempt from 'shared/hooks/useAttemptNext';
+import type { User } from 'teleport/services/user';
+import { DiscoverContext } from '../discoverContext';
+import type { AgentStepProps } from '../Shared';
+
+export function useLoginTrait({ ctx, props }: Props) {
+  const [user, setUser] = useState<User>();
+  const { attempt, run, setAttempt, handleError } = useAttempt('processing');
+  const [loginMap, setLoginMap] = useState<Record<string, boolean>>({});
```

Comment:
Can we use a map here? `Map<string, boolean>`

Reply:
After thinking about this, we might be able to use `Set<string>` instead - do we need to keep the entries where the value is `false`? If not, we can use a set and remove the value if it's set to false

Reply:
If you take my suggestion above then this becomes
```suggestion
  const [loginMap, setLoginMap] = useState<string[]>([]);
```